src/EnergyPlus/IceThermalStorage.cc

```diff
@@ -2408,172 +1710,75 @@ namespace IceThermalStorage {
     {
 
         Real64 CalcQstar;
-        
-        if (CurveIndVarType == CurveVarsFracChargedLMTD) {
-            CalcQstar = std::abs(CurveValue(CurveIndex, FracCharged, LMTDstar));
-        } else if (CurveIndVarType == CurveVarsFracDischargedLMTD) {
-            CalcQstar = std::abs(CurveValue(CurveIndex, (1.0-FracCharged), LMTDstar));
-        } else if (CurveIndVarType == CurveVarsLMTDMassFlow) {
-            CalcQstar = std::abs(CurveValue(CurveIndex, LMTDstar, MassFlowstar));
-        } else if (CurveIndVarType == CurveVarsLMTDFracCharged) {
-            CalcQstar = std::abs(CurveValue(CurveIndex, LMTDstar, FracCharged));
+
+        if (CurveIndVarType == CurveVars::FracChargedLMTD) {
+            CalcQstar = std::abs(CurveManager::CurveValue(CurveIndex, FracCharged, LMTDstar));
+        } else if (CurveIndVarType == CurveVars::FracDischargedLMTD) {
+            CalcQstar = std::abs(CurveManager::CurveValue(CurveIndex, (1.0 - FracCharged), LMTDstar));
+        } else if (CurveIndVarType == CurveVars::LMTDMassFlow) {
+            CalcQstar = std::abs(CurveManager::CurveValue(CurveIndex, LMTDstar, MassFlowstar));
+        } else if (CurveIndVarType == CurveVars::LMTDFracCharged) {
+            CalcQstar = std::abs(CurveManager::CurveValue(CurveIndex, LMTDstar, FracCharged));
         } else { // should never get here as this is checked on input
             CalcQstar = 0.0;
         }
-        
-        return CalcQstar;
 
+        return CalcQstar;
     }
 
-    
-    // *****************************************************************************
-
     Real64 TempSItoIP(Real64 const Temp)
     {
-
-        // Return value
-        Real64 TempSItoIP;
-
-        TempSItoIP = (Temp * 9.0 / 5.0) + 32.0;
-        return TempSItoIP;
+        return (Temp * 9.0 / 5.0) + 32.0;
```

Comment:
Hmm.  I'm definitely curious why we need to convert to IP internally.  But that's nothing to do with this refactor, so no worries.

Reply:
Yeah, there's some weird SI->IP->SI shenanigans going on in here.